clippy_lints/src/needless_for_each.rs

```diff
@@ -68,7 +68,9 @@ impl<'tcx> LateLintPass<'tcx> for NeedlessForEach {
             // e.g. `v.iter().for_each(f)` is simpler and clearer than using `for` loop.
             && let ExprKind::Closure(&Closure { body, .. }) = for_each_arg.kind
             && let body = cx.tcx.hir().body(body)
-            && let ExprKind::Block(..) = body.value.kind
+            // Skip the lint if the body is not safe, so as not to suggest `for … in … unsafe {}`
+            // and suggesting `for … in … { unsafe { } }` is a little ugly.
+            && let ExprKind::Block(Block { rules: BlockCheckMode::DefaultBlock, .. }, ..) = body.value.kind
```

Comment:
In case you ever want to support the `ugly` suggestion 😂, you may want to take `rules` out, then format the suggestion accordingly, such as:

```rust
if let BlockCheckMode::UnsafeBlock(_) = rules {
    Sugg::hir_with_applicability(cx, body.value, "..", &mut applicability).blockify()
} else {
    Sugg::hir_with_applicability(cx, body.value, "..", &mut applicability)
}
```

at line 102.

Which gives the correct suggestion

Reply:
`.blockify()`? huh.